### PR TITLE
fix up rkt libdir for stage1's

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

**Comment by mischief** [comment]
```
core@coreos_production_qemu-949-0-0-2 ~ $ sudo rkt trust --prefix=coreos.com/etcd --trust-keys-from-https
pubkey: prefix: "coreos.com/etcd"
key: "https://coreos.com/dist/pubkeys/aci-pubkeys.gpg"
gpg key fingerprint is: 8B86 DE38 890D DB72 9186  7B02 5210 BD88 8818 2190
        CoreOS ACI Builder <release@coreos.com>
Trusting "https://coreos.com/dist/pubkeys/aci-pubkeys.gpg" for prefix "coreos.com/etcd" without fingerprint review.
Added key for prefix "coreos.com/etcd" at "/etc/rkt/trustedkeys/prefix.d/coreos.com/etcd/8b86de38890ddb7291867b025210bd8888182190"
core@coreos_production_qemu-949-0-0-2 ~ $ sudo rkt run --stage1-from-dir=stage1-fly.aci coreos.com/etcd:v2.2.5
image: using image from file /usr/lib64/rkt/stage1-images/stage1-fly.aci
image: searching for app image coreos.com/etcd
image: remote fetching from URL "https://github.com/coreos/etcd/releases/download/v2.2.5/etcd-v2.2.5-linux-amd64.aci"
image: keys already exist for prefix "coreos.com/etcd", not fetching again
image: downloading signature from https://github.com/coreos/etcd/releases/download/v2.2.5/etcd-v2.2.5-linux-amd64.aci.asc
Downloading signature: [=======================================] 819 B/819 B
Downloading ACI: [=============================================] 7.73 MB/7.73 MB
image: signature verified:
  CoreOS ACI Builder <release@coreos.com>
run: can't evaluate mounts: missing volume for mountpoint "data-dir"
```

**Comment by mischief** [comment]
fixes coreos/rkt#2160

**Comment by marineam** [comment]
in the install section add a `dosym ../../$(get_libdir)/rkt/stage1-images/${aci} /usr/share/rkt/${aci}` since users may be referring to the old path.

File: app-emulation/rkt/rkt-9999.ebuild [hunk 1]
```diff
@@ -58,7 +58,7 @@ RDEPEND="!app-emulation/rocket
 
 BUILDDIR="build-${P}"
 
-STAGE1INSTALLDIR="/usr/share/rkt/"
+STAGE1INSTALLDIR="/usr/$(get_libdir)/rkt/stage1-images"
```
Comment (steveeJ) [inline app-emulation/rkt/rkt-9999.ebuild]: don't change this, look at the next comment

File: app-emulation/rkt/rkt-9999.ebuild [hunk 2]
```diff
@@ -92,7 +92,7 @@ src_configure() {
 	myeconfargs+=( $(use_enable tpm) )
 
 	myeconfargs+=( --with-stage1-flavors="${STAGE1FLAVORS}" )
-	myeconfargs+=( --with-stage1-default-location="/usr/share/rkt/stage1-${STAGE1FIRST}.aci" )
+	myeconfargs+=( --with-stage1-default-location="$(STAGE1INSTALLDIR)/stage1-${STAGE1FIRST}.aci" )
```
Comment (steveeJ) [inline app-emulation/rkt/rkt-9999.ebuild]: I think could should add `--with-stage1-default-images-directory=${STAGE1INSTALLDIR}` to change the lookup path

Comment (marineam) [inline app-emulation/rkt/rkt-9999.ebuild]: ${ not $(

Comment (mischief) [inline app-emulation/rkt/rkt-9999.ebuild]: the number of and effect of all the configure options in https://github.com/coreos/rkt/blob/master/Documentation/build-configure.md is quite baffling. :-/

**Comment by mischief** [comment]
@marineam ptal

File: app-emulation/rkt/rkt-9999.ebuild [hunk 3]
```diff
@@ -126,6 +126,9 @@ src_install() {
 		einfo $(basename "${stage1aci}")
 	done
 
+	# symlink old stage1 aci directory to the new install location
+	dosym ../../$(get_libdir)/rkt/stage1-images /usr/share/rkt
```
Comment (marineam) [inline app-emulation/rkt/rkt-9999.ebuild]: Extra ../ in there.

File: app-emulation/rkt/rkt-9999.ebuild (hunk 1 above)
Comment (marineam) [inline app-emulation/rkt/rkt-9999.ebuild]: I vote for changing to the rkt default, compatibility can be handled by a symlink.

File: app-emulation/rkt/rkt-9999.ebuild (hunk 3 above)
Comment (mischief) [inline app-emulation/rkt/rkt-9999.ebuild]: done

File: app-emulation/rkt/rkt-9999.ebuild [hunk 4]
```diff
@@ -126,6 +126,9 @@ src_install() {
 		einfo $(basename "${stage1aci}")
 	done
 
+	# symlink old stage1 aci directory to the new install location
+	dosym ../$(get_libdir)/rkt/stage1-images /usr/share/rkt
```
Comment (crawford) [inline app-emulation/rkt/rkt-9999.ebuild]: Is this necessary? When would a user have referenced this full path?

Comment (marineam) [inline app-emulation/rkt/rkt-9999.ebuild]: I don't know but I'm sure they did :-P

Comment (marineam) [inline app-emulation/rkt/rkt-9999.ebuild]: Not like the path was hidden, from https://coreos.com/rkt/

```
core@core-01 ~ $ sudo rkt run --interactive quay.io/coreos/alpine-sh -- /bin/sh
rkt: using image from file /usr/share/rkt/stage1-coreos.aci
rkt: using image from local store for image name quay.io/coreos/alpine-sh
/ # ps
```

Comment (marineam) [inline app-emulation/rkt/rkt-9999.ebuild]: Another google result: https://speakerdeck.com/joshix/coreos-rkt-fly?slide=8

:)

**Comment by marineam** [comment]
LGTM

File: app-emulation/rkt/rkt-9999.ebuild (hunk 4 above)
Comment (crawford) [inline app-emulation/rkt/rkt-9999.ebuild]: crap

**Pull request merged**